theano/configdefaults.py

```diff
@@ -1125,6 +1125,36 @@ def is_valid_check_preallocated_output_param(param):
              # Added elsewhere in the c key only when needed.
              in_c_key=False)
 
+if config.cxx == "icpc" and config.gcc.cxxflags:
```

Comment:
Take care with these checks since `config.cxx` is the full path to the compiler not just the executable name.  It's probably better to check for `'icpc' in config.cxx`